### ISSUE TITLE
Geom: Implement Splines, Rays and Polynomial hierarchy

### ISSUE THREAD

**Issue by scveloso** [UBC-Thunderbots/Software#957]
### Description of the task
Implement the following hierarchy as part of the geom redesign:
- Spline <- Line segment
- Ray
- Polynomial <- Line

These classes/interfaces should only have fields, getters and setters.
Methods that operate on these belong in util and they will be in a separate issue.

### Acceptance criteria

- [ ] Hierarchy implemented as above
- [ ] Each class has the fields, getters and setters it used to have before

### Blocked By
- #955
- #956

**Comment by garethellis0** [comment]
Just increasing the difficulty on this one a bit. We should probably doc the criteria for each difficulty level. Opened https://github.com/UBC-Thunderbots/Software/issues/1032 for myself to do that.

**Comment by jsn0716** [comment]
Only things left are contained in smaller issues #983 and #1098, closing this issue now
